src/YesSql.Core/Store.cs

```diff
@@ -341,7 +341,7 @@ internal IEnumerable<IndexDescriptor> CreateDescriptors(Type target, string coll
 
             foreach (var provider in indexProviders)
             {
-                if (provider.ForType().IsAssignableFrom(target) &&
+                if (provider.ForType().Equals(target) &&
```

Comment:
I'm not sure to be honest. 

I think from reading this that the `AnimalIndex` will never get written to if you're saving a `Dog`, which I think is what you want, but I'm not sure that it fits with all the other sub type arrangements within `YesSql`.

Have you had a read of the `ShouldQuerySubClasses()` tests and the behaviour that is expected of the actual document returned, rather than just what you can get in an index query?

Reply:
> I think from reading this that the AnimalIndex will never get written to if you're saving a Dog, which I think is what you want

exactly (animal and dog only have some parts in common, which doesn't mean to me writing one index should write another)

> Have you had a read of the ShouldQuerySubClasses() tests and the behaviour that is expected of the actual document returned, rather than just what you can get in an index query?

reading it now, so it looks like we have two different expected behaviors:
- one described by **ShouldQuerySubClasses**, where you can query base type and still expected to get documents of that base type and all others which inherited from base type,
- another one described by **ShouldRegisterIndexProviderWithModelInheritance**, where I tried to build two separated indexes (with inheritance) and expected that query over one index return some data and query over second index return other data.

That being said, I'm not sure if there is any chance to mix those two behaviors. Which means, that you can't build two **separate** indexes (one index derived from another).

I can live without this. Just need to duplicate some index fields.

Reply:
Now, I see that my example with **Animal** and **Dog** is not the best one, because you could expect saving dog will save animal too :-P

In my real world example, where I spotted this problem, I had one Index with some very common fields like PublishedUtc, CreatedUtc, and instead of duplicating those fields I wanted to derive from base index type.